### ISSUE TITLE
Safari – pasting text into title doesn't work as expected

### ISSUE THREAD

**Issue by marekzelezny** [WordPress/gutenberg#51047]
### Description

When pasting a text into title, the copied text is pasted in combination with a copy of title at the end of title. This happens only in Safari while using CMD+V. Using paste from menubar works fine.

### Step-by-step reproduction instructions

1. Open edit screen for any post
2. Use CMD+V to paste a text into title (like: 'test' between Hello and World)
3. Result is: _Hello world!testHello world!_

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/6258792/bb07ab19-3b4e-4707-b1cb-c10bddac9d10

### Environment info

- WordPress 6.2.2, Twenty Twenty-Three (clean install)
- No active plugins
- Safari 16.4

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by richtabor** [comment]
I've noticed this as well; even with selecting all/pasting a whole new title in.

**Comment by murshed** [comment]
Got the same on the post/page Title


![Image](https://github.com/WordPress/gutenberg/assets/437559/4f85e9de-eaf4-4c8d-8c3c-aae477dcd034)

**Comment by megane9988** [comment]
I got same.

- macOS 13.0（22A8380）
- Safari 16.1 (18614.2.9.1.12)

https://github.com/WordPress/gutenberg/assets/1205958/b519997b-3232-4a6b-8250-c1bf2583dc78

**Comment by lukasbesch** [comment]
I noticed the same issue. Additional information:

The cursor position before the paste does not matter, the result is always the same as described above.
After pasting, the cursor position is at the beginning (before the first letter).

My testing environment:
- macOS 12.6.5 (21G531)
- Safari 16.5 (17615.2.9.11.6, 17615)

**Comment by MatzeKitt** [comment]
It’s not only pasting, also copying doesn’t work as expected.

1. Write "This is a test" as title
2. Mark the word "test" by double-clicking on it
3. Press ⌘ + C to copy it
4. Paste the clipboard content with ⌘ + V inside the editor’s content (or anything else)

Expected: the word "test" is pasted.

Current behavior: The whole title "This is a test" is pasted.

You can also see via clipboard manager that the whole title is inside the clipboard manager instead of just the one word.

**Comment by Kuuuzya** [comment]
Hi everyone. Sorry, didn't noticed this ticket. If there any updates or fixes? Maybe someone know, how to fix it in Safari, cause using other browser on Mac is not to convenient :(

**Comment by ironprogrammer** [comment]
Today I dug a little deeper into this behavior, and it appears to affect WP back to 6.1. Not to say it's been an issue since the 6.1 release, but does affect the handling of <kbd>⌘+V</kbd> since [reported](https://github.com/WordPress/gutenberg/issues/51047#issue-1730451065) with Safari 16.4.

## Reproduction Report
### Environment
- Hardware: MacBook Pro Apple M1 Pro
- OS: macOS 14.4.1
- Browser: Safari 17.4.1, Google Chrome 123.0.6312.124, Mozilla Firefox 124.0.2
- PHP: 8.2.18
- MySQL: 8.0.27
- WordPress: 6.6-alpha-57778-src, 6.5.2, 6.0.2

### Actual Results
- Reproduction in Safari with:
	- ✅ `wordpress-develop:trunk` [Playground](https://playground.wordpress.net/?wp=nightly&url=/wp-admin/post-new.php) and local install
	- ✅ 6.5 - [Playground](https://playground.wordpress.net/?wp=6.5&url=/wp-admin/post-new.php) and local install
	- ✅ 6.5 - [Playground with Gutenberg](https://playground.wordpress.net/?wp=6.5&plugin=gutenberg&url=/wp-admin/post-new.php)
	- ✅ 6.4 - [Playground](https://playground.wordpress.net/?wp=6.4&url=/wp-admin/post-new.php)
	- ✅ 6.3 - [Playground](https://playground.wordpress.net/?wp=6.3&url=/wp-admin/post-new.php)
	- ✅ 6.2 - [Playground](https://playground.wordpress.net/?wp=6.2&url=/wp-admin/post-new.php)
	- ✅ 6.1 - [Playground](https://playground.wordpress.net/?wp=6.1&url=/wp-admin/post-new.php)
	- ❌ 6.0 local install

### Additional Notes
- It's notable that after pasting with  <kbd>⌘+V</kbd> in Safari, the carat is positioned at the _start of the title_. This is unexpected behavior for pasting operations, where the carat should be at the _end of the pasted text_.
- Could not reproduce in Chrome or Firefox.

### Supplemental Artifacts
Here are animated GIFs for tested versions:

<details>
<summary>6.5</summary>

![title-test-65](https://github.com/WordPress/gutenberg/assets/824344/53569bc7-33ff-47eb-bc22-fa1fc807acb3)

</details>

<details>
<summary>6.4</summary>

![title-test-64](https://github.com/WordPress/gutenberg/assets/824344/9687c705-6e7f-49fe-a50c-cda89d1a9da6)

</details>

<details>
<summary>6.3</summary>

![title-test-63](https://github.com/WordPress/gutenberg/assets/824344/02ea4759-60c6-4b35-b299-13220ac2b7a7)

</details>

<details>
<summary>6.2</summary>

![title-test-62](https://github.com/WordPress/gutenberg/assets/824344/ad71e7a4-46de-477c-bffd-bc94a3aab495)

</details>

<details>
<summary>6.1</summary>

![title-test-61](https://github.com/WordPress/gutenberg/assets/824344/df36c64e-4164-401a-9520-39167a1da03f)

</details>

**Comment by Mamaduka** [comment]
When pasting in the Safari post title, I noticed that the RichText value's `start` and `end` properties are `undefined`. This is the reason for unexpected paste behavior. Emulating the same in Chrome will give us the same results:

It can be run via the DevTools console in the editor

```js
const value = { ...wp.richText.create( { html: 'Hey' } ), start: undefined, end: undefined };
const newValue = wp.richText.insert( value, wp.richText.create( { html: ' There' } ) );

// { text: 'Hey ThereHey' }
console.log( newValue );
```

## Screenshot
![CleanShot 2024-08-21 at 06 49 52](https://github.com/user-attachments/assets/d334de2e-4473-47df-9d81-8fac2d6eb853)

**Comment by Mamaduka** [comment]
Here's the fix - #64671.
